### PR TITLE
enhance git cheatsheet

### PR DESCRIPTION
**Pull request by papanito**
- add more info for work with branches
- add more info for work with tags
- add info about gpgsign

### CODE REVIEW

File: git [hunk 1]
```diff
@@ -55,7 +55,10 @@ git filter-branch --env-filter \
 git reset --hard
 
 # To go 2 commits back:
-git reset --hard HEAD~2
+git reset --hard HEAD~
```
Comment (papanito) [inline git:58]: re-add `2`

**Comment by papanito** [comment]
Ready to be merged

**Review by thehappydinoa — APPROVED** [review]

**Comment by chrisallenlane** [comment]
Sorry for the delay here - this all looks great. I just haven't had time to review this at-length yet (which I try to do for all cheatsheet PRs). I'll get to this this weekend for sure :slightly_smiling_face:

**Comment by papanito** [comment]
@chrisallenlane sure no issue, appreciate your time and effort.

**Comment by chrisallenlane** [comment]
Merged, finally. I'm sorry it took such an unreasonable length of time to get to this. At this point, I really should have learned to strike "for sure" from my vocabulary.

**Pull request merged**